projects/igniteui-angular/src/lib/grids/api.service.ts

```diff
@@ -158,17 +148,17 @@ export class GridBaseAPIService <T extends IgxGridBaseDirective & GridType> {
     update_cell(cell: IgxCell, value: any) {
         cell.editValue = value;
         const args = cell.createEditEventArgs();
-        this.grid.onCellEdit.emit(args);
-        // TODO Implement cellEditExit event end emit if isEqual(args.oldValue, args.newValue)
-        // TODO do not emit onCellEdit & cellEditDone if isEqual(args.oldValue, args.newValue)
-        if (args.cancel) {
+
+        if (isEqual(args.oldValue, args.newValue)) {
             return args;
         }
 
-        if (isEqual(args.oldValue, args.newValue)) {
+        this.grid.cellEdit.emit(args);
+        if (args.cancel) {
             return args;
         }
 
+
```

Comment:
Can't leave a comment after `this.grid.cellEditDone.emit(doneArgs);` (line 181), but:
The `IgxCell` object on the service never existed for more than a single edit until now, but now if you stop the cell exit it will. Problem is its `value` is only assigned once and every consecutive edit operation will have wrong old value argument or can fail the equals check at the start here when it shouldn't. Something like:
```ts
cell.value = args.newValue;
```
after emitting done should fix that